packages/boot/tools/ibc/mocks.js

```diff
@@ -1,3 +1,48 @@
+/** @import { ChannelId, ChannelOrdering, ConnectionId, DestinationPort, SourcePort } from '@agoric/orchestration'; */
+
+/**
+ * based on BridgeId.DIBC events
+ * @typedef {{
+ *  hops: ConnectionId[];
+ *  method: 'startChannelOpenInit';
+ *  order: ChannelOrdering;
+ *  packet: {
+ *    destination_port: DestinationPort;
+ *    source_port: SourcePort;
+ *  };
+ *  type: 'IBC_METHOD';
+ *  version: string;
+ * }} ChannelOpenInitPacket
+ */
+
+/**
+ * based on BridgeId.DIBC events
+ * @typedef {{
+ *  method: 'sendPacket';
+ *  packet: {
+ *    data: string;
+ *    destination_channel: ChannelId;
+ *    destination_port: DestinationPort;
+ *    source_channel: ChannelId;
+ *    source_port: SourcePort;
+ *  };
+ *  relativeTimeoutNs: string;
+ *  type: 'IBC_METHOD';
+ * }} SendPacket
+ */
+
+/**
+ * @param {string} versionString version string
```

Comment:
please convey that this string itself must be in JSON

Reply:
It's an arbitrary string.  ICA happens to use a JSON format, but IBC transfer uses a non-JSON string of just seven bytes:  `ics20-1`.

I'm increasingly hesitant to impose more structure than the IBC spec describes (more than just ICA or the "IBC transfer" protocol).  There's a sweet spot to be found between making APIs with useful guard rails versus types that are flexible enough to express legitimate uses of IBC we haven't yet imagined.

Reply:
> impose more structure than the IBC spec describes

💯 we should stick to the spec. Are there types for the spec that we could import? If not, I propose we have a separate file with just the types from the spec that code implementing parts of it imports.

But for parts of the code that do not support the full spec, shouldn't we document what they are expecting?